### PR TITLE
RFR: Automated global labels configuration test cases

### PR DESCRIPTION
**Pull request by ganeshhubale**
Automated global labels configuration test cases  
  - test_crud_global_custom_label
  - test_search_global_custom_label
  - test_analysis_global_custom_label
  - test_invalid_label_file_type
  - test_total_global_system_label
  - test_search_global_system_label

Created common collection for both label and rules configurations
  - Collection added - GlobalCollection
  - Added global_configuration.py file
  - This file contains all views and navigations regitration for rules and labels

- Updated rules test cases as per this collection

Signed-off-by: Ganesh Hubale <ghubale@redhat.com>

### CODE REVIEW

File: mta/entities/analysis_results.py [hunk 1]
```diff
@@ -73,6 +73,8 @@ def is_displayed(self):
 class AnalysisDetailsView(BaseLoggedInPage):
     """This view represents details page of specific analysis"""
 
+    title = Text(".//h1")
```
Comment (sshveta) [inline mta/entities/analysis_results.py:76]: no text present ?

File: mta/entities/global_configuration.py [hunk 2]
```diff
@@ -152,33 +153,170 @@ def is_displayed(self):
...
+    table = Table(
+        locator='.//table[contains(@aria-label, "Table")]',
+        column_widgets={
+            "Provider ID": Button(locator='.//th[@data-label="Provider ID"]/button'),
+            2: Dropdown(),
+        },
+    )
+    search = Input(locator='.//input[@aria-label="Filter by provider ID"]')
+
+    def search_project(self, provider_id):
```
Comment (sshveta) [inline mta/entities/global_configuration.py:170]: @ganeshhubale : We should not be adding methods like search , delete etc to View classes . A collection is needed when we can create more than one object of anything like in Projects , we can create more than one project . SO we can have a collection for projects. 

In system rules we are not creating anything just validating things so we don't need Global collection . 

You have register your navigations like 
      https://github.com/windup/windup_integration_test/blob/master/mta/entities/analysis_results.py#L147

**Review by sshveta — CHANGES_REQUESTED** [review]
@ganeshhubale : I suggest 
1) creating separate classes for rules and labels 
2) We don't need Global collection , see analysis result class for example 
3) Methods like search , delete will come under Class SystemRules() not views .
We can discuss more if you have questions

File: mta/entities/analysis_results.py (hunk 1 above)
Comment (ganeshhubale) [reply mta/entities/analysis_results.py:76]: Nope. there is no `text` attribute inside `h1` tag.

File: mta/entities/global_configuration.py (hunk 2 above)
Comment (ganeshhubale) [reply mta/entities/global_configuration.py:170]: I have followed `AnalysisResults` way and improved automation.

**Comment by ganeshhubale** [comment]
All tests are passing locally
```
(.mta_env) [ghubale@localhost windup_integration_test]$ py.test mta/tests/test_global_labels.py -vvv
============================================================================ test session starts ============================================================================
platform linux -- Python 3.7.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /home/ghubale/migration_toolkit_for_applications/windup_integration_test/.mta_env/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/ghubale/migration_toolkit_for_applications/windup_integration_test
plugins: rerunfailures-9.1.1, polarion-collect-0.25.0, mta-integration-tests-0.1.dev200+gcfccf97.d20210330
collected 6 items                                                                                                                                                           

mta/tests/test_global_labels.py::test_crud_global_custom_label PASSED                                                                                                 [ 16%]
mta/tests/test_global_labels.py::test_search_global_custom_label PASSED                                                                                               [ 33%]
mta/tests/test_global_labels.py::test_analysis_global_custom_label PASSED                                                                                             [ 50%]
mta/tests/test_global_labels.py::test_invalid_label_file_type PASSED                                                                                                  [ 66%]
mta/tests/test_global_labels.py::test_total_global_system_label PASSED                                                                                                [ 83%]
mta/tests/test_global_labels.py::test_search_global_system_label PASSED                                                                                               [100%]

============================================================================= warnings summary ==============================================================================
```

```
(.mta_env) [ghubale@localhost windup_integration_test]$ py.test mta/tests/test_global_rules.py -vvv
============================================================================ test session starts ============================================================================
platform linux -- Python 3.7.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /home/ghubale/migration_toolkit_for_applications/windup_integration_test/.mta_env/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/ghubale/migration_toolkit_for_applications/windup_integration_test
plugins: rerunfailures-9.1.1, polarion-collect-0.25.0, mta-integration-tests-0.1.dev200+gcfccf97.d20210330
collected 7 items                                                                                                                                                           

mta/tests/test_global_rules.py::test_crud_global_custom_rule PASSED                                                                                                   [ 14%]
mta/tests/test_global_rules.py::test_search_global_custom_rule PASSED                                                                                                 [ 28%]
mta/tests/test_global_rules.py::test_analysis_global_custom_rule PASSED                                                                                               [ 42%]
mta/tests/test_global_rules.py::test_invalid_rule_file_type PASSED                                                                                                    [ 57%]
mta/tests/test_global_rules.py::test_applied_global_custom_rule PASSED                                                                                                [ 71%]
mta/tests/test_global_rules.py::test_total_global_system_rule PASSED                                                                                                  [ 85%]
mta/tests/test_global_rules.py::test_filter_global_system_rule PASSED                                                                                                 [100%]

============================================================================= warnings summary ==============================================================================```

**Comment by ganeshhubale** [comment]
Updated code as per discussion.
All tests are passing
```
(.mta_env) [ghubale@localhost windup_integration_test]$ py.test mta/tests/test_global_*.py -vvv
============================================================================ test session starts ============================================================================
platform linux -- Python 3.7.5, pytest-6.2.1, py-1.10.0, pluggy-0.13.1 -- /home/ghubale/migration_toolkit_for_applications/windup_integration_test/.mta_env/bin/python3.7
cachedir: .pytest_cache
rootdir: /home/ghubale/migration_toolkit_for_applications/windup_integration_test
plugins: rerunfailures-9.1.1, polarion-collect-0.25.1.dev0+g37f7ba9.d20210330, mta-integration-tests-0.1.dev201+g6913019.d20210331
collected 13 items                                                                                                                                                          

mta/tests/test_global_labels.py::test_crud_global_custom_label PASSED                                                                                                 [  7%]
mta/tests/test_global_labels.py::test_search_global_custom_label PASSED                                                                                               [ 15%]
mta/tests/test_global_labels.py::test_analysis_global_custom_label PASSED                                                                                             [ 23%]
mta/tests/test_global_labels.py::test_invalid_label_file_type PASSED                                                                                                  [ 30%]
mta/tests/test_global_labels.py::test_total_global_system_label PASSED                                                                                                [ 38%]
mta/tests/test_global_labels.py::test_search_global_system_label PASSED                                                                                               [ 46%]
mta/tests/test_global_rules.py::test_crud_global_custom_rule PASSED                                                                                                   [ 53%]
mta/tests/test_global_rules.py::test_search_global_custom_rule PASSED                                                                                                 [ 61%]
mta/tests/test_global_rules.py::test_analysis_global_custom_rule PASSED                                                                                               [ 69%]
mta/tests/test_global_rules.py::test_invalid_rule_file_type PASSED                                                                                                    [ 76%]
mta/tests/test_global_rules.py::test_applied_global_custom_rule PASSED                                                                                                [ 84%]
mta/tests/test_global_rules.py::test_total_global_system_rule PASSED                                                                                                  [ 92%]
mta/tests/test_global_rules.py::test_filter_global_system_rule PASSED                                                                                                 [100%]

============================================================================= warnings summary ==============================================================================
mta/tests/test_global_labels.py: 5 warnings
mta/tests/test_global_rules.py: 7 warnings
  /home/ghubale/migration_toolkit_for_applications/windup_integration_test/.mta_env/lib64/python3.7/site-packages/selenium/webdriver/remote/webelement.py:695: DeprecationWarning: encodestring() is a deprecated alias since 3.1, use encodebytes()
    content = base64.encodestring(fp.getvalue())

-- Docs: https://docs.pytest.org/en/stable/warnings.html
================================================================ 13 passed, 12 warnings in 736.57s (0:12:16) ================================================================
(.mta_env) [ghubale@localhost windup_integration_test]$ 
```

**Review by sshveta — APPROVED** [review]

**Pull request merged**